tests/sources/test_redis.py

```diff
@@ -335,3 +335,34 @@ async def test_advanced_rules_validation(advanced_rules, expected_validation_res
             advanced_rules
         )
         assert validation_result == expected_validation_result
+
+
+@pytest.mark.asyncio
+async def test_ping_when_mutual_ssl_enabled():
+    async with create_redis_source() as source:
+        source.client.database = ["*"]
+        source.client.tls_enabled = True
+        source.client.mutual_tls_enabled = True
+        source.client.tls_certfile = (
+            "-----BEGIN CERTIFICATE----- Invalid-Certificate -----END CERTIFICATE-----"
+        )
+        source.client.tls_keyfile = "-----BEGIN RSA PRIVATE KEY----- Invalid-Certificate -----END RSA PRIVATE KEY-----"
+        mocked_client = AsyncMock()
+        with mock.patch(
+            "redis.from_url",
+            return_value=mocked_client,
+        ):
+            await source.client._client
```

Comment:
Probably you want to assert 
```python
mocked_client.assert_called_once_with(<connection_string>, decode_responses=True)
```

BTW, you don't use `await` here as `_client` is not a coroutine

Reply:
mocked_client.assert_called_once_with(<connection_string>, decode_responses=True)
This line will be called 0 times which fails the test case. instead of that I added `hasaatr` assert for testing.

![Screenshot 2024-02-08 144800](https://github.com/elastic/connectors/assets/125074214/275d0e4b-5804-4506-892b-93a8d9462514)

Reply:
My bad, I think I gave the wrong suggestion.

You should assert `redis.from_url` is called with the expected arguments.

Reply:
The patch of `redis.from_url` won't work as you used the alias here:
https://github.com/elastic/connectors/blob/814678802b88c70ca5a4c5e47dc9c2e931d7a056/connectors/sources/redis.py#L11


Try below test:
```python
@pytest.mark.asyncio
async def test_client_when_mutual_ssl_enabled():
    async with create_redis_source() as source:
        source.client.database = ["*"]
        source.client.ssl_enabled = True
        source.client.mutual_tls_enabled = True
        source.client.store_ssl_key = Mock(return_value = "/tmp/tmp4vn0jxy7.crt")
        with mock.patch(
                "redis.asyncio.from_url",
                return_value=AsyncMock(),
        ) as from_url_mock:
            redis_client = source.client._client
            connection_string = "rediss://username:password@localhost:6379?ssl_certfile=/tmp/tmp4vn0jxy7.crt&ssl_keyfile=/tmp/tmp4vn0jxy7.crt"
            from_url_mock.assert_called_with(connection_string, decode_responses=True)
```